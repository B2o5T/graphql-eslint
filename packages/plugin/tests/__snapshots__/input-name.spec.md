// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Code

      1 | type Mutation { SetMessage(message: String): String }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/2

    > 1 | type Mutation { SetMessage(message: String): String }
        |                            ^^^^^^^ Input \`message\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { SetMessage(input: String): String }

❌ Error 2/2

    > 1 | type Mutation { SetMessage(message: String): String }
        |                                     ^^^^^^ Input type \`String\` name should be \`SetMessageInput\`.

💡 Suggestion: Rename to \`SetMessageInput\`

    1 | type Mutation { SetMessage(message: SetMessageInput): String }
`;

exports[` 2`] = `
⚙️ Options

    {
      "checkInputType": true
    }

❌ Error

    > 1 | type Mutation { SetMessage(input: String): String }
        |                                   ^^^^^^ Input type \`String\` name should be \`SetMessageInput\`.

💡 Suggestion: Rename to \`SetMessageInput\`

    1 | type Mutation { SetMessage(input: SetMessageInput): String }
`;

exports[` 3`] = `
⚙️ Options

    {
      "checkInputType": true
    }

❌ Error

    > 1 | type Mutation { SetMessage(hello: SetMessageInput): String }
        |                            ^^^^^ Input \`hello\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { SetMessage(input: SetMessageInput): String }
`;

exports[` 4`] = `
Code

      1 | type Mutation { userCreate(record: CreateOneUserInput!): CreateOneUserPayload }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/2

    > 1 | type Mutation { userCreate(record: CreateOneUserInput!): CreateOneUserPayload }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: CreateOneUserInput!): CreateOneUserPayload }

❌ Error 2/2

    > 1 | type Mutation { userCreate(record: CreateOneUserInput!): CreateOneUserPayload }
        |                                    ^^^^^^^^^^^^^^^^^^ Input type \`CreateOneUserInput\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: userCreateInput!): CreateOneUserPayload }
`;

exports[` 5`] = `
Code

      1 | type Mutation { userCreate(record: [CreateOneUserInput]!): CreateOneUserPayload }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput]!): CreateOneUserPayload }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: [CreateOneUserInput]!): CreateOneUserPayload }

❌ Error 2/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput]!): CreateOneUserPayload }
        |                                     ^^^^^^^^^^^^^^^^^^ Input type \`CreateOneUserInput\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: [userCreateInput]!): CreateOneUserPayload }
`;

exports[` 6`] = `
Code

      1 | type Mutation { userCreate(record: [CreateOneUserInput!]!): CreateOneUserPayload }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput!]!): CreateOneUserPayload }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: [CreateOneUserInput!]!): CreateOneUserPayload }

❌ Error 2/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput!]!): CreateOneUserPayload }
        |                                     ^^^^^^^^^^^^^^^^^^ Input type \`CreateOneUserInput\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: [userCreateInput!]!): CreateOneUserPayload }
`;

exports[` 7`] = `
Code

      1 | type Mutation { userCreate(record: [CreateOneUserInput!]): CreateOneUserPayload }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput!]): CreateOneUserPayload }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: [CreateOneUserInput!]): CreateOneUserPayload }

❌ Error 2/2

    > 1 | type Mutation { userCreate(record: [CreateOneUserInput!]): CreateOneUserPayload }
        |                                     ^^^^^^^^^^^^^^^^^^ Input type \`CreateOneUserInput\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: [userCreateInput!]): CreateOneUserPayload }
`;

exports[` 8`] = `
Code

      1 | type Mutation { userCreate(record: String, test: String): String }

⚙️ Options

    {
      "checkInputType": true
    }

❌ Error 1/4

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: String, test: String): String }

❌ Error 2/4

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                                    ^^^^^^ Input type \`String\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: userCreateInput, test: String): String }

❌ Error 3/4

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                                            ^^^^ Input \`test\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(record: String, input: String): String }

❌ Error 4/4

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                                                  ^^^^^^ Input type \`String\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(record: String, test: userCreateInput): String }
`;

exports[` 9`] = `
Code

      1 | type Mutation { userCreate(record: String, test: String): String }

⚙️ Options

    {
      "checkInputType": false
    }

❌ Error 1/2

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                            ^^^^^^ Input \`record\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(input: String, test: String): String }

❌ Error 2/2

    > 1 | type Mutation { userCreate(record: String, test: String): String }
        |                                            ^^^^ Input \`test\` should be called \`input\`.

💡 Suggestion: Rename to \`input\`

    1 | type Mutation { userCreate(record: String, input: String): String }
`;

exports[` 10`] = `
⚙️ Options

    {
      "checkInputType": true,
      "caseSensitiveInputType": false
    }

❌ Error

    > 1 | type Mutation { userCreate(input: String): String }
        |                                   ^^^^^^ Input type \`String\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(input: userCreateInput): String }
`;

exports[` 11`] = `
⚙️ Options

    {
      "checkInputType": true,
      "caseSensitiveInputType": true
    }

❌ Error

    > 1 | type Mutation { userCreate(input: UserCreateInput): String }
        |                                   ^^^^^^^^^^^^^^^ Input type \`UserCreateInput\` name should be \`userCreateInput\`.

💡 Suggestion: Rename to \`userCreateInput\`

    1 | type Mutation { userCreate(input: userCreateInput): String }
`;

exports[` 12`] = `
⚙️ Options

    {
      "checkQueries": true,
      "checkInputType": true,
      "caseSensitiveInputType": true
    }

❌ Error

    > 1 | type Query { getUser(input: GetUserInput): String }
        |                             ^^^^^^^^^^^^ Input type \`GetUserInput\` name should be \`getUserInput\`.

💡 Suggestion: Rename to \`getUserInput\`

    1 | type Query { getUser(input: getUserInput): String }
`;
