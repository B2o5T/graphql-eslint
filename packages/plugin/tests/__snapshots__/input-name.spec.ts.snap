// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Error 1/2

> 1 | type Mutation { SetMessage(message: String): String }
    |                            ^^^^^^^ Input "message" should be called "input"

Error 2/2

> 1 | type Mutation { SetMessage(message: String): String }
    |                                     ^^^^^^ InputType "String" name should be "SetMessageInput"
`;

exports[` 2`] = `
Error 1/1

> 1 | type Mutation { SetMessage(input: String): String }
    |                                   ^^^^^^ InputType "String" name should be "SetMessageInput"
`;

exports[` 3`] = `
Error 1/1

> 1 | type Mutation { SetMessage(hello: SetMessageInput): String }
    |                            ^^^^^ Input "hello" should be called "input"
`;

exports[` 4`] = `
Error 1/2

> 1 | type Mutation { userCreate(record: CreateOneUserInput!): CreateOneUserPayload }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/2

> 1 | type Mutation { userCreate(record: CreateOneUserInput!): CreateOneUserPayload }
    |                                    ^^^^^^^^^^^^^^^^^^ InputType "CreateOneUserInput" name should be "userCreateInput"
`;

exports[` 5`] = `
Error 1/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput]!): CreateOneUserPayload }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput]!): CreateOneUserPayload }
    |                                     ^^^^^^^^^^^^^^^^^^ InputType "CreateOneUserInput" name should be "userCreateInput"
`;

exports[` 6`] = `
Error 1/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput!]!): CreateOneUserPayload }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput!]!): CreateOneUserPayload }
    |                                     ^^^^^^^^^^^^^^^^^^ InputType "CreateOneUserInput" name should be "userCreateInput"
`;

exports[` 7`] = `
Error 1/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput!]): CreateOneUserPayload }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/2

> 1 | type Mutation { userCreate(record: [CreateOneUserInput!]): CreateOneUserPayload }
    |                                     ^^^^^^^^^^^^^^^^^^ InputType "CreateOneUserInput" name should be "userCreateInput"
`;

exports[` 8`] = `
Error 1/4

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/4

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                                    ^^^^^^ InputType "String" name should be "userCreateInput"

Error 3/4

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                                            ^^^^ Input "test" should be called "input"

Error 4/4

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                                                  ^^^^^^ InputType "String" name should be "userCreateInput"
`;

exports[` 9`] = `
Error 1/2

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                            ^^^^^^ Input "record" should be called "input"

Error 2/2

> 1 | type Mutation { userCreate(record: String, test: String): String }
    |                                            ^^^^ Input "test" should be called "input"
`;

exports[` 10`] = `
Error 1/1

> 1 | type Mutation { userCreate(input: String): String }
    |                                   ^^^^^^ InputType "String" name should be "userCreateInput"
`;

exports[` 11`] = `
Error 1/1

> 1 | type Mutation { userCreate(input: UserCreateInput): String }
    |                                   ^^^^^^^^^^^^^^^ InputType "UserCreateInput" name should be "userCreateInput"
`;

exports[` 12`] = `
Error 1/1

> 1 | type Query { getUser(input: GetUserInput): String }
    |                             ^^^^^^^^^^^^ InputType "GetUserInput" name should be "getUserInput"
`;
